command-expansion-server/src/lib/codegen/CommandEDSLPreface.ts

```diff
@@ -2,46 +2,122 @@
+  public readonly relativeTime?: Temporal.Duration;
+  public readonly epochTime?: { epochName: string; time: Temporal.Duration };
 
-  private constructor(opts: { stem: string; arguments: A }) {
+  private constructor(opts: {
+    stem: string;
+    arguments: A;
+    absoluteTime?: Temporal.Duration;
+    relativeTime?: Temporal.Duration;
+    epochTime?: { epochName: string; time: Temporal.Duration };
+  }) {
     this.stem = opts.stem;
     this.arguments = opts.arguments;
+    this.absoluteTime = opts.absoluteTime;
+    this.relativeTime = opts.relativeTime;
+    this.epochTime = opts.epochTime;
   }
 
-  public static new<A extends any[] | { [argName: string]: any }>(opts: { stem: string; arguments: A }): Command<A> {
+  public static new<A extends any[] | { [argName: string]: any }>(opts: {
+    stem: string;
+    arguments: A;
+    absoluteTime?: Temporal.Duration;
+    relativeTime?: Temporal.Duration;
+    epochTime?: { epochName: string; time: Temporal.Duration };
+  }): Command<A> {
     return new Command({
       stem: opts.stem,
       arguments: opts.arguments,
+      absoluteTime: opts.absoluteTime,
+      relativeTime: opts.relativeTime,
+      epochTime: opts.epochTime,
     });
   }
 
   public toSeqJson() {
-    return {
+    let seqJson: any = {
       id: 'command',
       metadata: {},
+
       steps: [
         {
           stem: this.stem,
-          time: { type: 'COMPLETE' },
+          time: {
+            tag: this.absoluteTime
+              ? this.absoluteTime.toString()
+              : this.relativeTime
+              ? this.relativeTime.toString()
+              : this.epochTime
+              ? this.epochTime.time.toString()
+              : '',
+            type: this.absoluteTime
+              ? 'ABSOLUTE'
+              : this.relativeTime
+              ? 'COMMAND_RELATIVE'
+              : this.epochTime
+              ? 'EPOCH_RELATIVE'
+              : 'COMMAND_COMPLETE',
+          },
           type: 'command',
           metadata: {},
           args: typeof this.arguments == 'object' ? Object.values(this.arguments) : this.arguments,
         },
       ],
     };
+    // include epoch field if we have an epoch time
+    if (this.epochTime) {
+      seqJson['epoch'] = this.epochTime.epochName;
+    }
+
+    return seqJson;
+  }
+
+  public absoluteTiming(absoluteTime: Temporal.Duration): Command<A> {
+    return new Command({
+      stem: this.stem,
+      arguments: this.arguments,
+      absoluteTime: absoluteTime,
+    });
+  }
+
+  public relativeTiming(relativeTime: Temporal.Duration): Command<A> {
+    return new Command({
+      stem: this.stem,
+      arguments: this.arguments,
+      relativeTime: relativeTime,
+    });
+  }
+
+  public epochTiming(epochName: string, epochTime: Temporal.Duration): Command<A> {
+    return new Command({
+      stem: this.stem,
+      arguments: this.arguments,
+      epochTime: { epochName: epochName, time: epochTime },
+    });
```

Comment:
In case you are wondering why I am creating a new command object instead of doing the below:
```
public absoluteTiming(absoluteTime: Temporal.Duration): Command<A> {
    this.absoluteTime = absoluteTime;
    return this;
  }
```

If we are using a command multiple times you will override the values or store multiple time values in the single command.


In this example:
```
// My instrument will prepare 2 bread loafs

PREPARE_BREAD
BAKE_BREAD.absoluteTiming(Temporal.Duration.from({ minutes: 5 })),
....
....
PREPARE_BREAD
BAKE_BREAD.absoluteTiming(Temporal.Duration.from({ minutes: 15 })),
```

Both BAKE_BREAD will generate an absolute time of 15 minutes, not 5 and 15

Reply:
Ah yes, this is a limitation of how we are skipping parens for commands without arguments by making it a single exported singleton. This could be sufficient cause to reverse that decision and require the parens.